mautic/mautic
# Fix API Permission Issue

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

It is the API Permission Issue while fetching form submissions. If user do not have view form submissions permission. It was returning 500 error code. It should return the proper error message ("You do not have access to the requested area/action") with 403 error code.


#### Steps to reproduce the bug:
1. Remove Form permissions, like ‘view own’ and ‘view others’
2. Make request to fetch form submissions (api/forms/<form_id>/submissions)
3. Observe 500 error messages in logs.


#### Steps to test this PR:


1. Load up [this PR](https://mautibox.com)
2. Follow steps to reproduce the bug
3. Observe the proper error message : “You do not have access to the requested area/action”




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10679"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
